src/app/components/kudos-user-history/kudos-user-history.component.js

```diff
@@ -0,0 +1,229 @@
+                else if (self.challengeOperations) updateListChallenges(self.thisUsersEmail, false);
+            }
+
+            if (val == 'less') {
+                self.transactionEndingIndex = 6;
+                self.showMoreButton = true;
+                self.showLessButton = false;
+            }
+        }
+        
+        function showList(email, trigger) {
+            self.transactionEndingIndex = 6;
+
+            if (self.allOperations) updateListAll(email, trigger);
+            else if (self.receivedOperations) updateListReceived(email, trigger);
+            else if (self.gaveOperations) updateListGave(email, trigger);
+            else if (self.challengeOperations) updateListChallenges(email, trigger);
+        }
+
+        function changeRadioValue(value) {
+            if (self.historyRadioBox != value) {
+                self.historyRadioBox = value;
+            }
+        }
+
+        function changeView(value) {
+            if (value == 'all') {
+                self.receivedOperations = false;
+                self.allOperations = true;
+                self.gaveOperations = false;
+                self.challengeOperations = false;
+            } else if (value == 'received'){
+                self.receivedOperations=true;
+                self.allOperations=false;
+                self.gaveOperations=false;
+                self.challengeOperations=false;
+            } else if (value == 'gave'){
+                self.receivedOperations=false;
+                self.allOperations=false;
+                self.gaveOperations=true;
+                self.challengeOperations=false;
+            } else if (value == 'challenges'){
+                self.receivedOperations=false;
+                self.allOperations=false;
+                self.gaveOperations=false;
+                self.challengeOperations=true;
+            }
+
+            self.showLessButton = false;
+            self.transactionEndingIndex = 6;
+            self.historyRadioBox = 'wonLost';
+        }
+
+        function getTemplate(page) {
+            if (this.page == true){
+                return 'app/components/kudos-user-history/kudos-user-history-page.html';
+            } else {
+                return 'app/components/kudos-user-history/kudos-user-history.html';
+            }
+        }
+
+        function spin() {
+            $timeout(function () {
+                self.spin = false;
+            }, 1000);
+        }
+
+    };
+
+    UserHistoryController.$inject = ['Resources', 'UserHistoryService', '$timeout'];
+
+    angular.module('myApp.components.userHistory', [])
+        .component('kudosUserHistory', {
+            template: '<ng-include src="self.getTemplate()"/>' ,
+            bindings: {
+                user: '=',
+                page: '<'
+            },
+            controller: 'UserHistoryController',
+            controllerAs: 'self'
+        })
+        .controller('UserHistoryController', UserHistoryController)
```

Comment:
Single responsibility principle - one component per file. I'd have no problem if the controller was declared via function, but here you are declaring a new component.